zerver/lib/test_helpers.py

```diff
@@ -90,6 +91,45 @@ def stub_event_queue_user_events(
             yield
 
 
+class activate_push_notification_service(override_settings):  # noqa: N801
+    """
+    Activating the push notifications service involves a few different settings
+    that are logically related, and ordinarily set correctly in computed_settings.py
+    based on the admin-configured settings.
+    Having tests deal with overriding all the necessary settings every time they
+    want to simulate using the push notification service would be too
+    cumbersome, so we provide a convenient helper.
+    Can be used as either a context manager or a decorator applied to a test method
+    or class, just like original override_settings.
+    """
+
+    def __init__(
+        self, zulip_services_url: Optional[str] = None, submit_usage_statistics: bool = False
+    ) -> None:
+        if zulip_services_url is None:
+            zulip_services_url = settings.ZULIP_SERVICES_URL
+        assert zulip_services_url is not None
+
+        zulip_services = settings.ZULIP_SERVICES or set()
+        zulip_services.add("mobile_push")
+
+        if settings.ANALYTICS_DATA_UPLOAD_LEVEL < AnalyticsDataUploadLevel.BILLING:
+            analytics_data_upload_level = AnalyticsDataUploadLevel.BILLING
+        else:  # nocoverage
+            analytics_data_upload_level = settings.ANALYTICS_DATA_UPLOAD_LEVEL
+
+        if submit_usage_statistics:
+            zulip_services.add("submit_usage_statistics")
+            analytics_data_upload_level = AnalyticsDataUploadLevel.ALL
+
+        super().__init__(
+            ZULIP_SERVICES_URL=zulip_services_url,
+            ZULIP_SERVICES=zulip_services,
+            ANALYTICS_DATA_UPLOAD_LEVEL=analytics_data_upload_level,
+            USE_PUSH_NOTIFICATIONS_SERVICE=True,
```

Comment:
Noticed that I'm not being consistent in naming settings/functions/etc.  `push notificationS service` (plural) or `push notification service`. Any preference one way or another, aside of fixing this to be consistent?

Reply:
Probably worth asking in `#backend` for visibility; I'm not sure what's best.

Reply:
updated the changes here to be consistent and use "push notification service", since that's what matches the official name. there are other inconsistencies throughout the code, but that's out of scope of this PR and there's a different discussion about that.

Reply:
Sounds good!